docs/applications/nublado/major-upgrade.rst

```diff
@@ -0,0 +1,313 @@
+
+Turn off ``mobu`` if you're using it, so that it doesn't complain
+about the service outage.
+
+Disable logins
+""""""""""""""
+
+Disable user logins to nublado v2 and nublado v3.  This is most easily
+done by deleting the ``hub`` deployment from each of the ``nublado2``
+and ``nublado`` applications.
+
+Remove user sessions
+""""""""""""""""""""
+
+Remove any user sessions.  This is not, strictly speaking, necessary,
+but if you have a user with a running installation who decides to use
+the other endpoint, they will risk the file corruption mentioned above.
+You can do this by deleting the namespace objects in the
+``nublado-users`` application (which will also clean up anything within
+those namespaces) or by executing something like ``for i in '' 2; do
+kubectl get ns | grep nublado${i}- | awk '{print $1}' | xargs kubectl
+delete ns; done``.
+
+Bring up applications
+"""""""""""""""""""""
+#. Synchronize the ``nublado2`` application.  This will move the
+   default spawner to the ``/n2`` route.
+#. Test that you can get a lab and that that lab allows use of the
+   command line and cell executions with the new spawner at ``/n2``.  If
+   you find errors, troubleshoot them.  If nublado v2 was working
+   correctly beforehand, there should be no errors.
+#. Synchronize the ``nublado`` application.  This will turn on the
+   nublado v3 spawner on the default route ``/nb``.  At this point users
+   will start returning (whether or not you call an all-clear, which you
+   should not yet; users don't pay attention to such things).
+#. Re-enable mobu, if you disabled it.  Wait to make sure it's correctly
+   running notebooks, or test a few yourself.
+
+Remove nublado v2
+-----------------
+
+Pre-deployment Preparation
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+As before, these steps can be done at any time, and should be done in
+advance of the maintenance window.
+
+Communicate to user base
+""""""""""""""""""""""""
+
+Let your users know that ``/n2`` will no longer available.  Announce that
+that maintenance window will be disruptive and will destroy any user
+sessions spawned via ``/n2``.  Hopefully that will be none or at least
+very few.
+
+Turn off nublado v2
+"""""""""""""""""""
+
+Delete the entries ``cachemachine: true``, ``moneypenny: true``, and
+``nublado2: true`` from your instance-specific values file in
+``environment``.
+
+Deployment
+^^^^^^^^^^
+
+During the deployment window:
+
+#. Synchronize the ``science-platform`` application.  This will delete
+   the ``nublado2``, ``cachemachine``, and ``moneypenny`` applications.
+#. Remove any user sessions spawned by ``nublado2``.  This is harder to
+   do from the UI: you will need to find any users in ``nublado-users``
+   with a namespace called ``nublado2-<username>`` and remove only those
+   namespaces.  From the CLI, it's easier: ``kubectl get ns | grep
+   nublado2- | awk '{print $1}' | xargs kubectl delete ns``.
+#. Delete the values files for your installation from Phalanx's
```

Comment:
for your environment

Reply:
I actually wouldn't do any of this deleting until you know that the new version is good, maybe break that into some kind of cleanup section - the instructions make sense

Reply:
yeah I'd put the deleting things into the everyone is running it section.  You never know when usdf is going to find some problem.